src/main/java/org/ecocean/Decision.java

```diff
@@ -165,6 +164,103 @@ public static String findWinner(JSONObject winningIndividualTracker) {
+      System.out.println("deleteMe getNumberOfVotesForMostAgreedUponMatchesInParallelOrder called");
+      List<Integer> numberOfVotes = new ArrayList<Integer>();
+      String currentMatchedMarkedIndividualId = null;
+      Double currentMatchedMarkedIndividualCounter = 0.0;
+      JSONObject winningIndividualTracker = new JSONObject();
+      JSONObject currentDecisionValue = new JSONObject();
+      if (decisionsForEncounter != null && decisionsForEncounter.size() > 0) {
+        for (Decision currentDecision : decisionsForEncounter) {
+          if (currentDecision.getProperty().equals("match")) {
+            currentDecisionValue = currentDecision.getValue();
+            currentMatchedMarkedIndividualId = currentDecisionValue
+                .optString("id", null);
+            currentMatchedMarkedIndividualCounter = winningIndividualTracker
+                .optDouble(currentMatchedMarkedIndividualId, 0.0);
+            winningIndividualTracker.put(currentMatchedMarkedIndividualId,
+                currentMatchedMarkedIndividualCounter + 1);
+          }
+        }
+        numberOfVotes = sortVotesByPopularity(winningIndividualTracker);
+      }
+      System.out.println("deleteMe returning numberOfVotes as : " + numberOfVotes.toString());
+      return numberOfVotes;
+    }
+
+    public static List<Integer> sortVotesByPopularity(JSONObject winningIndividualTracker) { //TODO DRY up
+      List<Integer> votes = new ArrayList<Integer>();
+      int minVotesToBeIncluded = 1;
+      Iterator<String> keys = winningIndividualTracker.keys();
+      String key = null;
+      while (keys.hasNext()) {
+        key = keys.next();
+        if (winningIndividualTracker.optInt(key, 0) >= minVotesToBeIncluded) {
+          votes.add(winningIndividualTracker.optInt(key, 0));
+        }
+      }
+      Collections.sort(votes);
+      Collections.reverse(votes);
+      return votes;
+    }
+
+    public static List<String> sortIdsByPopularity(JSONObject winningIndividualTracker){
+      List<String> resultsArr = new ArrayList<String>();
+      List<Integer> votes = new ArrayList<Integer>();
+      List<String> idsInParallelWitVotes = new ArrayList<String>();
+      int minVotesToBeIncluded = 1;
+      Iterator<String> keys = winningIndividualTracker.keys();
+      String key = null;
+      while(keys.hasNext()) {
+          key = keys.next();
+          if(winningIndividualTracker.optInt(key,0) >= minVotesToBeIncluded){
+            votes.add(winningIndividualTracker.optInt(key,0));
+            idsInParallelWitVotes.add(key);
+          }
+      }
+      final List<String> stringListCopy = new ArrayList(idsInParallelWitVotes);
+      ArrayList<String> sortedList = new ArrayList(stringListCopy);
+      int[] votesPrimitive = convertIntegers(votes);
+      Collections.sort(sortedList, Comparator.comparing(s -> votesPrimitive[stringListCopy.indexOf(s)])); //ugh votesPrimitive array needs to be primitive gross.
```

Comment:
this is a pretty decent intro to doing custom comparators in java -- i always have to look these things up; but i think a lot of this sorting you are doing could be done with a sub-class in this class just for the sake of tallying/sorting votes.  i would have to think it over a bit more myself... but just mentioning it for _future_ experimentation for such cases.  this can stay like it is, i am sure.  especially since the list is small and we dont worry too much about efficiency.  basically rather than using JSONObject to do the tallying, you can just make a special class like VoteScore that you could pass in Decisions.  the comparator would sort those.  something like that. heh